### PR TITLE
feat(docs): add gpu user guide

### PR DESCRIPTION
**Pull request by Major-333**
This PR adds a comprehensive GPU User Guide to docs, enabling NVIDIA GPU support in Docker and Kubernetes. It provides step-by-step instructions, code snippets, and YAML configs for easy implementation. Now users can seamlessly utilize GPU resources for compute-intensive workloads.

### CODE REVIEW

**Comment by Antlera** [comment]
Should we make the gpu guide a separate guidance document? If so, we need to replace [GPU guide in developer_guide](https://github.com/intelligent-machine-learning/dlrover/blob/master/docs/developer_guide.md#minikube-with-gpu-support) with a reference to this guide.

File: docs/tutorial/gpu_user_guide.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+```
+
+This will install the necessary components for NVIDIA GPU support in Docker, enabling you to utilize GPU resources within Docker containers.
+
+#### Step 3: Set NVIDIA as the default runtime for Docker
+
+To configure Docker to use NVIDIA as the default runtime, follow these steps:
+
+1. Edit the Docker daemon configuration file:
+
```
Comment (Antlera) [inline docs/tutorial/gpu_user_guide.md:32]: `$ sudo nvidia-ctk runtime configure --runtime=docker --set-as-default` can also be used with configuration in one step.

File: docs/tutorial/gpu_user_guide.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+```
+
+2. Add the following content to the file:
+
+```json
+{
+    "default-runtime": "nvidia",
+    "runtimes": {
+        "nvidia": {
+            "path": "nvidia-container-runtime",
```
Comment (Antlera) [inline docs/tutorial/gpu_user_guide.md:44]: In my case, k8s failed to collect `nvidia-container-runtime`, if the `path` is not in absolute form.
Which is to say, the `/etc/docker/daemon.json` should be as followed:
```yaml
#-------/etc/docker/daemon.json
{
    "default-runtime": "nvidia",
    "runtimes": {
        "nvidia": {
            "args": [],
            "path": "/usr/bin/nvidia-container-runtime"
        }
    }
}
```

File: docs/tutorial/gpu_user_guide.md (hunk 1 above)
Comment (Major-333) [reply docs/tutorial/gpu_user_guide.md:32]: thx

File: docs/tutorial/gpu_user_guide.md (hunk 2 above)
Comment (Major-333) [reply docs/tutorial/gpu_user_guide.md:44]: It could be related to the environment variable $PATH. I couldn't reproduce the issue on my end.

**Comment by Major-333** [comment]
> Should we make the gpu guide a separate guidance document? If so, we need to replace [GPU guide in developer_guide](https://github.com/intelligent-machine-learning/dlrover/blob/master/docs/developer_guide.md#minikube-with-gpu-support) with a reference to this guide.

Indeed, making the GPU guide a separate document would add clarity and detail. We can certainly replace the GPU guide in the developer_guide with a reference to this new document.

File: docs/tutorial/gpu_user_guide.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
+### GPU User Guide
+
+#### Step 1: Prepare the system for NVIDIA GPU support
+
+To enable NVIDIA GPU support, follow these steps to configure the system:
+
+```bash
+distribution=$(. /etc/os-release;echo $ID$VERSION_ID) \
```
Comment (workingloong) [inline docs/tutorial/gpu_user_guide.md:8]: Where do we run the command?

Comment (Major-333) [reply docs/tutorial/gpu_user_guide.md:8]: The first four steps in this documentation need to be run on each bare-metal machine that requires GPU usage.

Comment (Major-333) [reply docs/tutorial/gpu_user_guide.md:8]: > Where do we run the command?
The first four steps in this documentation need to be run on each bare-metal machine that requires GPU usage.

Comment (Major-333) [reply docs/tutorial/gpu_user_guide.md:8]: > Where do we run the command?

The first four steps in this documentation need to be run on each bare-metal machine that requires GPU usage.

Comment (Major-333) [reply docs/tutorial/gpu_user_guide.md:8]: Let me add this note to the documentation：
> "The first four steps in this document need to be run on each bare-metal machine that will use a GPU. If you've already set up each node that requires GPU usage, or you're working in a well-maintained cloud-based Kubernetes environment, you can directly start from step five."

**Review by workingloong — APPROVED** [review]
LGTM

**Pull request merged**